File: src/HMS.Domain/Entities/LabTest.cs
using HMS.Domain.Common;

namespace HMS.Domain.Entities;

public class LabTest : BaseEntity
{
    public Guid PatientId { get; set; }
    public Patient Patient { get; set; } = null!;

    public Guid? MedicalRecordId { get; set; }
    public MedicalRecord? MedicalRecord { get; set; }

    public string TestName { get; set; } = string.Empty;
    public string? TestType { get; set; }
    public DateTime RequestedDate { get; set; } = DateTime.UtcNow;
    public DateTime? ResultDate { get; set; }
    public string? ResultValue { get; set; }
    public string? NormalRange { get; set; }
    public string? Remarks { get; set; }
    public bool IsCompleted { get; set; } = false;
}

File: frontend/src/types/labTest.ts
export interface LabTestDto {
  id: string;
  patientId: string;
  patientName: string;
  medicalRecordId: string | null;
  testName: string;
  testType: string | null;
  requestedDate: string;
  resultDate: string | null;
  resultValue: string | null;
  normalRange: string | null;
  remarks: string | null;
  isCompleted: boolean;
  createdAt: string;
}

export interface CreateLabTestDto {
  patientId: string;
  medicalRecordId?: string;
  testName: string;
  testType?: string;
  requestedDate: string;
}

export interface UpdateLabTestDto {
  testName: string;
  testType?: string;
  requestedDate: string;
  resultDate?: string;
  resultValue?: string;
  normalRange?: string;
  remarks?: string;
  isCompleted: boolean;
}
